test/framework/src/main/java/org/elasticsearch/test/hamcrest/ElasticsearchAssertions.java

```diff
@@ -636,7 +636,7 @@ public static void assertVersionSerializable(Version version, Streamable streama
          * a way that sucks less.
          */
         NamedWriteableRegistry registry;
-        if (ESIntegTestCase.isInternalCluster()) {
+        if (ESIntegTestCase.isInternalCluster() && ESIntegTestCase.internalCluster().size() > 0) {
```

Comment:
This is mostly used to make sure that all communications between the embedded nodes serialize at some random version. Since you can only communicate if there is a node to communicate with, I'm somewhat surprised you got here. The condition makes sense and I'm fine with it, but it seems curious that you'd make it here. I wonder if there is some race condition that never comes up in CI but you can trigger it by stepping through certain code.

Either way, I'm ok with the addition here.